pkg/storage/s3storage.go

```diff
@@ -0,0 +1,518 @@
+// Fork from Thanos S3 Bucket support to reuse configuration options
+// Licensed under the Apache License 2.0
+// https://github.com/thanos-io/thanos/blob/main/pkg/objstore/s3/s3.go
```

Comment:
@michaelmdresser @AjayTripathy This file is forked from the Thanos S3 store implementation, mainly because we're using the Thanos bucket config format for all our bucket secret configuration. Thanos uses Apache 2 License (same as cost-model). I'm just doing my due diligence here to make sure this is all we need in order to meet the license requirements, or is there is further action required?

Reply:
Forking from Apache2.0 is fine.